Automattic/wp-calypso
# Checkout: refactor Jetpack pre-purchase notices

#### Changes proposed in this Pull Request

* Create a `PrePurchaseNotices` component responsible for determining which (if any) notices should be shown during checkout. This component looks at the state of the current cart, rather than solely the current product being checked out.
* Use `PrePurchaseNotices` in `CheckoutSystemDecider`, replacing the existing notice logic there.
* Move three existing pre-purchase notices into the `prepurchase-notices` directory, as a step toward consolidating all notices in one place.
* Rename notices for more clarity:
  * `IncludedProductNoticeContent` ➡️ `SitePlanIncludesCartProductNotice`
  * `OwnedProductNoticeContent` ➡️ `CartPlanOverlapsOwnedProductNotice`
  * `JetpackMinimumPluginVersionNoticeContent` ➡️ `JetpackPluginRequiredVersionNotice`
* Create a `PrePurchaseNotice` component (to be used within the `prepurchase-notices` directory) that renders a "standard" notice with an optional call-to-action link.
* Refactor the three above-mentioned notices to use the `PrePurchaseNotice` component.

#### Testing instructions

* Test that notices still show up as expected when the cart includes a Backup product that is covered by a site's existing plan. (For more info, see #44873.)
* Test that notices still show up as expected when the cart includes a plan that overlaps with Backup products already purchased for a site. (For more info, see #44873.)
* Test that notices still show up as expected for sites that need to upgrade their plugin to support Backup. (For more info, see #44886.)

#### Reference screenshots

##### The current site plan includes a product in the cart

<img width="566" alt="image" src="https://user-images.githubusercontent.com/670067/90265375-1d471300-de18-11ea-8438-005ff78ad19f.png">

##### The plan in the cart includes a product already owned

<img width="571" alt="image" src="https://user-images.githubusercontent.com/670067/90265579-626b4500-de18-11ea-994d-e66c5f3f53d9.png">

##### The site's Jetpack plugin requires an update to support Jetpack Backup

<img width="567" alt="image" src="https://user-images.githubusercontent.com/670067/90265464-4071c280-de18-11ea-9fba-1557cca053a2.png">
